pkg/aws/awsclient.go

```diff
@@ -0,0 +1,173 @@
+package aws
+
+import (
+	"context"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	"github.com/aws/aws-sdk-go-v2/service/sqs"
+	"github.com/aws/aws-sdk-go-v2/service/iam"
+	"github.com/aws/aws-sdk-go-v2/service/eks"
+	"github.com/aws/aws-sdk-go-v2/service/pricing"
+	"github.com/aws/aws-sdk-go-v2/service/ssm"
+	"github.com/aws/aws-sdk-go-v2/service/cloudformation"
+	"github.com/aws/aws-sdk-go-v2/service/timestreamwrite"
+	"github.com/aws/aws-sdk-go-v2/service/fis"
+	"github.com/aws/aws-sdk-go-v2/service/sts"
+)
+
+type AWSClient struct {
```

Comment:
So we don't have to group everything into a single client -- I think we can just have separate interfaces per-service -- that'll keep things well-defined enough but still flexible for testing